Recrutify/Recrutify.Host/Infrastructure/Authorization/RolesPolicyHandler.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Newtonsoft.Json;
+
+namespace Recrutify.Host.Infrastructure.Authorization
+{
+    public class RolesPolicyHandler : AuthorizationHandler<RolesPolicyRequirement>
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public RolesPolicyHandler(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        protected override Task HandleRequirementAsync(AuthorizationHandlerContext context, RolesPolicyRequirement requirement)
+        {
+            var projectId = _httpContextAccessor?.HttpContext.Request.Query[Constants.Roles.ProjectIdParam].ToString();
+            var projectRoles = context.User.Claims
+                .Where(c => c.Type == Constants.Roles.Role)
+                .Select(c => JsonConvert.DeserializeObject<ProjectRoles>(c.Value))
+                .ToDictionary(c => c.ProjectId, c => c.Roles);
+
+            if (projectRoles.TryGetValue(Guid.Parse(projectId ??= DataAccess.Constants.Roles.GlobalProjectId.ToString()), out var globalRolesValue)
```

Comment:
?? should be on line 21, and Guid.Parse too, here you should pass only projectId